### PR TITLE
Notification Update

### PR DESCRIPTION
**Pull request by Maan-Sharma**
Create a Notification.kt file to mannage the Notification push

### CODE REVIEW

File: app/src/main/java/com/example/cartoon/Notification.kt [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+                Manifest.permission.POST_NOTIFICATIONS
+            ) != PackageManager.PERMISSION_GRANTED
+        ) {
+            FancyToast.makeText(
+                context,
+                "No image selected", FancyToast.LENGTH_LONG,
+                FancyToast.WARNING,true).show()
+
+        }
+        notificationManager.notify(notificationId, notification)
```
Comment (lingarajsankaravelu) [inline app/src/main/java/com/example/cartoon/Notification.kt:43]: Hey @Maan-Sharma . You should check the notification permission in activity itself before calling sendNotification. Reason being. 
1. check the notification permission -> register activity for result - > you will know whether user have given permission or not. Take action based on granted or not granted. 


You can refer this. https://developer.android.com/training/permissions/requesting

**Review by lingarajsankaravelu — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/cartoon/Notification.kt (hunk 1 above)
Comment (lingarajsankaravelu) [reply app/src/main/java/com/example/cartoon/Notification.kt:43]: @ashut0sh75 . I think this change needed here. Can cause a crash in android 13 and above. due to the notification being triggered without user permission.

Comment (ashut0sh75) [reply app/src/main/java/com/example/cartoon/Notification.kt:43]: Thanks for pointing it out @lingarajsankaravelu . I tried the app in 13 and we saw that notification is not generated also I think there was an issue with toast being triggered unnecessarily. 
Should we ask for users permissions separately to resolve it ?

Comment (lingarajsankaravelu) [reply app/src/main/java/com/example/cartoon/Notification.kt:43]: @ashut0sh75 . Yes users permission should be from activity.

Comment (ashut0sh75) [reply app/src/main/java/com/example/cartoon/Notification.kt:43]: okk sure.

**Pull request merged**